scipy/stats/_stats_py.py

```diff
@@ -293,9 +307,13 @@ def hmean(a, axis=0, dtype=None, *, weights=None):
     if weights is not None:
         weights = np.asarray(weights, dtype=dtype)
 
-    if not np.all(a >= 0):
-        message = ("The harmonic mean is only defined if all elements are greater "
-                   "than or equal to zero; otherwise, the result is NaN.")
+    negative_mask = a < 0
```

Comment:
The different is in NaN handling. Typically, NaNs in the input do not produce NaNs in the output, so `np.any(a < 0)` is preferable to (and easier to read than) `not np.all(a >= 0)`. In practice, this doesn't matter much because the decorator already handles the case of `NaN` input.